config/application.rb

```diff
@@ -43,5 +43,8 @@ class Application < Rails::Application
       config.logstasher.logger_path = LOGSTASH_PATH
       config.logstasher.log_controller_parameters = true
     end
+
+    # Set the host and protocol for all action mailer urls
+    config.action_mailer.default_url_options = { :host => SERVER_URL, :protocol => SERVER_PROTOCOL }
```

Comment:
I think this would be better in `config/initializers/action_mailer.rb` if possible. I like to restruct changes to `application.rb` to what can't go elsewhere as it makes updates easier.

Reply:
Done. There's an order-of-loading issue with using `Rails.application.config` to achieve this in the initializer, so I just went with overriding ActionMailer::Base.default_url_options instead. That seems reasonable and matches the smtp_settings approach too.